src/displayapp/screens/settings/SettingSetDateTime.cpp

```diff
@@ -0,0 +1,240 @@
+      case 11:
+        return 30;
+      default:
+        return 31;
+    }
+  }
+}
+
+bool SettingSetDateTime::OnTouchEvent(Pinetime::Applications::TouchEvents event) {
+  return screens.OnTouchEvent(event);
+}
+
+SettingSetDateTime::SettingSetDateTime(Pinetime::Applications::DisplayApp* app,
+                                       Pinetime::Controllers::DateTime& dateTimeController,
+                                       Pinetime::Controllers::Settings& settingsController)
+  : Screen(app),
+    dateTimeController {dateTimeController},
+    settingsController {settingsController},
+    screens {app,
+             0,
+             {[this]() -> std::unique_ptr<Screen> {
+                return CreateScreen1();
+              },
+              [this]() -> std::unique_ptr<Screen> {
+                return CreateScreen2();
+              }},
+             Screens::ScreenListModes::UpDown} {
+}
+
+std::unique_ptr<Screen> SettingSetDateTime::CreateScreen1() {
+
+  lv_obj_t* title = lv_label_create(lv_scr_act(), nullptr);
+  lv_label_set_text_static(title, "Set current date");
+  lv_label_set_align(title, LV_LABEL_ALIGN_CENTER);
+  lv_obj_align(title, lv_scr_act(), LV_ALIGN_IN_TOP_MID, 15, 15);
+
+  lv_obj_t* icon = lv_label_create(lv_scr_act(), nullptr);
+  lv_obj_set_style_local_text_color(icon, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_ORANGE);
+
+  lv_label_set_text_static(icon, Symbols::clock);
+  lv_label_set_align(icon, LV_LABEL_ALIGN_CENTER);
+  lv_obj_align(icon, title, LV_ALIGN_OUT_LEFT_MID, -10, 0);
+
+  dayCounter.SetValueChangedEventCallback(this, DateValueChangedHandler);
+  dayCounter.Create();
+  dayCounter.SetValue(dateTimeController.Day());
+  lv_obj_align(dayCounter.GetObject(), nullptr, LV_ALIGN_CENTER, POS_X_DAY, POS_Y_TEXT);
+
+  monthCounter.EnableMonthMode();
+  monthCounter.SetValueChangedEventCallback(this, DateValueChangedHandler);
+  monthCounter.Create();
+  monthCounter.SetValue(static_cast<int>(dateTimeController.Month()));
+  lv_obj_align(monthCounter.GetObject(), nullptr, LV_ALIGN_CENTER, POS_X_MONTH, POS_Y_TEXT);
+
+  yearCounter.SetValueChangedEventCallback(this, DateValueChangedHandler);
+  yearCounter.Create();
+  yearCounter.SetValue(dateTimeController.Year());
+  lv_obj_align(yearCounter.GetObject(), nullptr, LV_ALIGN_CENTER, POS_X_YEAR, POS_Y_TEXT);
+
+  btnSetDate = lv_btn_create(lv_scr_act(), nullptr);
+  btnSetDate->user_data = this;
+  lv_obj_set_size(btnSetDate, 120, 48);
+  lv_obj_align(btnSetDate, lv_scr_act(), LV_ALIGN_IN_BOTTOM_MID, 0, 0);
+  lv_obj_set_style_local_bg_color(btnSetDate, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_MAKE(0x38, 0x38, 0x38));
+  lblSetDate = lv_label_create(btnSetDate, nullptr);
+  lv_label_set_text_static(lblSetDate, "Set");
+  lv_obj_set_event_cb(btnSetDate, SetDateEventHandler);
+  lv_btn_set_state(btnSetDate, LV_BTN_STATE_DISABLED);
+  lv_obj_set_state(lblSetDate, LV_STATE_DISABLED);
+
+  return std::make_unique<Screens::DotLabel>(0, 2, app, title);
+}
```

Comment:
A tidier solution would probably be to keep the original SetDate and SetTime files/screens, and simply use them in the new screenlist like so.

```suggestion
std::unique_ptr<Screen> SettingSetDateTime::CreateScreen1() {
  return std::make_unique<Screens::SettingSetDate>(...);
}
```

Reply:
I disagree, I think it should all be in one file. Both SetDate and SetTime are shown as one screen, thus they should be found in one file. Also getting back setDate and setTime isnt going to be easy because of how I merged two branches to combine the files and retain the git history, if I did it get them back it would be really messy. Therefore this PR would need to be scrapped. Having three different files for this would just be too complex for something this simple in my opinion. 

additionally, I dont exactly how your suggestion would work...lol

Reply:
Files don't add complexity, they create organization. I don't see a good reason to combine two classes that are completely independent from each other into one, where half the functions and variables relate to one screen, and the other half to a different one.

Reply:
You can always clean up the history with git-rebase, force pushing, or we can just squash merge this PR.

Reply:
Ok then.... could you elaborate on how that would work? especially what goes into the (...) part. I usually figure things out by  looking how other apps work but there isnt really anything similar to that :/

Reply:
The parameters of the constructor for that screen.

https://github.com/InfiniTimeOrg/InfiniTime/blob/afea7ca0d1d670bdee04cfe80a1d8c36efa4fca0/src/displayapp/DisplayApp.cpp#L413
https://github.com/InfiniTimeOrg/InfiniTime/blob/afea7ca0d1d670bdee04cfe80a1d8c36efa4fca0/src/displayapp/screens/settings/SettingSetDate.h#L14

Reply:
Ok, that makes much more sense. Thanks!